apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/Transaction.java

```diff
@@ -83,7 +83,7 @@ public Transaction(ElasticApmTracer tracer) {
 
     public Transaction start(@Nullable String traceParentHeader, long startTimestampNanos, Sampler sampler) {
         if (traceParentHeader != null) {
-            traceContext.asChildOf(traceParentHeader);
+            traceContext.asChildOf(traceParentHeader, sampler);
```

Comment:
Your call of course, but it might be a bit neater to leave the concern of what to do in case of an invalid traceparent header to the caller. e.g. you could make `asChildOf` return a bool indicating success, and then:

```java
if (traceParentHeader == null || !traceContext.asChildOf(traceParentHeader)) {
    traceContext.asRootSpan(sampler);
}
```

Reply:
nice idea!